service/history/replication/stream_receiver.go

```diff
@@ -150,9 +144,9 @@ func (r *StreamReceiver) sendEventLoop() {
 		select {
 		case <-timer.C:
 			timer.Reset(r.Config.ReplicationStreamSyncStatusDuration())
-			streamCreationTime, stream := r.getStream()
-			if err := r.ackMessage(stream); err != nil {
-				r.recreateStream(streamCreationTime)
+			if err := r.ackMessage(r.stream); err != nil {
+				r.logger.Error("StreamReceiver exit send loop", tag.Error(err))
+				return
```

Comment:
the solution is to not recreate the gRPC stream, rely on the stream monitor to recreate the entire thing